src/Spryker/Zed/SynchronizationBehavior/Persistence/Propel/Behavior/SynchronizationBehavior.php

```diff
@@ -1197,4 +1200,129 @@ public function getTableOrFail(): Table
 
         return $table;
     }
+
+    /**
+     * @return int
+     */
+    protected function isDirectSyncPerEntityDisabled(): int
+    {
+        return isset($this->getParameters()['direct_sync_disabled']) ? 1 : 0;
+    }
+
+    /**
+     * @return string
+     */
+    protected function addIsDirectSyncEnabledMethod(): string
+    {
+        $isDirectSynchronizationEnabled = $this->getConfig()->isDirectSynchronizationEnabled() ? 1 : 0;
+        $isDirectSyncPerEntityDisabled = $this->isDirectSyncPerEntityDisabled();
+
+        return "
+/**
+ * @return int
+ */
+protected function isDirectSyncEnabled(): int
```

Comment:
```suggestion
protected function isDirectSyncEnabled(): bool
```

Reply:
After transformation to behavior function will return `false -> '' or true -> 1`, so boolean can't be achieved here.